src/utilities/do-try.test.ts

```diff
@@ -0,0 +1,190 @@
+            // Arrange
+            const workload = async () => {
+                throw Error();
+            };
+
+            // Act & Assert
+            Do.try(workload).catch(
+                (result?: ResultRecord<any>, error?: any) => {
+                    expect(error).not.toBeNull();
+                    expect(error).not.toBeUndefined();
+                    expect(result).toBeUndefined();
+                }
+            );
+            expect.assertions(3);
+        });
+
+        it("When no errors occur, then catch handler is never called", async () => {
+            // Arrange
+            const catchHandler = jest.fn();
+            const workload = jest.fn();
+
+            // Act
+            await Do.try(workload)
+                .catch(catchHandler)
+                .getAwaiter();
+
+            // Assert
+            expect(catchHandler).not.toHaveBeenCalled();
+        });
+
+        it("When no errors occur, then finally handler is still called", async () => {
+            // Arrange
+            const finallyHandler = jest.fn();
+            const workload = jest.fn();
+
+            // Act
+            await Do.try(workload)
+                .finally(finallyHandler)
+                .getAwaiter();
+
+            // Assert
+            expect(finallyHandler).toHaveBeenCalled();
+        });
+
+        it("When errors occur, then finally handler is still called", async () => {
+            // Arrange
+            const catchHandler = jest.fn();
+            const workload = async () => {
+                throw Error();
+            };
+            let finallyCalled = false;
+
+            // Act & Assert
+            Do.try(workload)
+                .catch(catchHandler)
+                .finally(() => {
+                    finallyCalled = true;
+                    expect(finallyCalled).toBeTrue();
+                });
+        });
+    });
+
+    describe("DoSync.try", () => {
+        it("When no errors occur, then .execute() returns the workload return value", () => {
+            // Arrange
+            const workload = () => StubResourceRecordFactory.build();
+
+            // Act
+            const result = DoSync.try(workload).execute();
+
+            // Assert
+            expect(result).not.toBeUndefined();
+            expect(result).not.toBeNull();
+            expect(result).toBeInstanceOf(StubResourceRecord);
+        });
+
+        it("When Javascript errors occur, then catch handler is passed regular error and return value is undefined", () => {
+            // Arrange
+            const workload = () => {
+                throw Error();
+            };
+
+            // Act
+            const result = DoSync.try(workload)
+                .catch((result?: ResultRecord<any>, error?: any) => {
+                    expect(result).toBeUndefined();
+                    expect(error).not.toBeUndefined();

```

Comment:
Minor, but there is a `toBeNil()` matcher which could reduce the assertions here. That should handle `undefined` or `null`. Not a big deal, I don't think we've been using it, but I was pretty sure there was a matcher that did both 😛

Reply:
hm, I'm not seeing that matcher anywhere.

Reply:
ah, looks like it does exist, just VS Code intellisense doesn't pick it up.

Reply:
Yeah, we still need to iron out the typings for `jest-extended`